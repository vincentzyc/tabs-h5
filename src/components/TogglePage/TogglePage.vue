<template>
  <div class="change-button-wrap">
    <div class="flex flex-center">
      <van-button
        @click="getPrePage()"
        class="change-button pre-button"
        icon="arrow-left"
        round
        type="primary"
        :disabled="preDisable"
        >上一页</van-button
      >
      <van-button
        @click="getNextPage()"
        class="change-button next-button"
        icon="arrow"
        icon-position="right"
        loading-text="下一页"
        round
        type="primary"
        :loading="nextLoading"
        >下一页</van-button
      >
    </div>
  </div>
</template>

<script setup lang="ts">
const props = defineProps<{
  pageIndex: number;
  nextLoading: boolean;
}>();

const emits = defineEmits(["prepage", "nextpage"]);

const preDisable = $computed(() => props.pageIndex === 1);

function getPrePage() {
  emits("prepage");
}
function getNextPage() {
  emits("nextpage");
}
</script>
