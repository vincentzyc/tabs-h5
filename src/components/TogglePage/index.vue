<template>
  <div class="change-button-wrap">
    <div class="flex flex-center">
      <van-button @click="getPrePage()" class="change-button" icon="arrow-left" round type="primary"
        >上一页</van-button
      >
      <van-button
        @click="getNextPage()"
        class="change-button"
        icon="arrow"
        icon-position="right"
        loading-text="下一页"
        round
        type="primary"
        >下一页</van-button
      >
    </div>
  </div>
</template>

<script setup lang="ts">
function getPrePage() {
  console.log("上一页");
}
function getNextPage() {
  console.log("下一页");
}
</script>
