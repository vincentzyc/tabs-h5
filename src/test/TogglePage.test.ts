import { mount } from '@vue/test-utils'
import TogglePage from '@/components/TogglePage/TogglePage.vue'

test('mount TogglePage component', async () => {
  expect(TogglePage).toBeTruthy()

  const wrapper = mount(TogglePage, {
    props: {
      pageIndex:1,
      nextLoading:false
    },
  })

  await wrapper.get('.change-button').trigger('click')
  
  const emitEvent = wrapper.emitted('prepage')

  expect(emitEvent).toHaveLength(1)
})
