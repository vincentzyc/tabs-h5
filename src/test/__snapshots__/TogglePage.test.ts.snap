// Vitest Snapshot v1

exports[`mount TogglePage component 1`] = `
"<div class=\\"change-button-wrap\\">
  <div class=\\"flex flex-center\\"><button type=\\"button\\" class=\\"van-button van-button--primary van-button--normal van-button--round van-button--disabled change-button pre-button\\" disabled=\\"\\">
      <div class=\\"van-button__content\\"><i class=\\"van-badge__wrapper van-icon van-icon-arrow-left van-button__icon\\">
          <!---->
          <!---->
          <!---->
        </i><span class=\\"van-button__text\\">上一页</span>
        <!---->
      </div>
    </button><button type=\\"button\\" class=\\"van-button van-button--primary van-button--normal van-button--round change-button next-button\\">
      <div class=\\"van-button__content\\">
        <!----><span class=\\"van-button__text\\">下一页</span><i class=\\"van-badge__wrapper van-icon van-icon-arrow van-button__icon\\">
          <!---->
          <!---->
          <!---->
        </i>
      </div>
    </button></div>
</div>"
`;
